src/freenas/usr/local/lib/middlewared_truenas/plugins/failover.py

```diff
@@ -1108,32 +1108,47 @@ def upgrade_finish(self, job):
 
     @private
     async def sync_keys_to_remote_node(self, lock=True):
+        """
+        Sync GELI and/or ZFS encryption keys to the standby node.
+        """
+
         if await self.middleware.call('failover.licensed'):
-            async with ENCRYPTION_CACHE_LOCK if lock else asyncnullcontext():
-                try:
-                    keys = await self.encryption_keys()
-                    await self.middleware.call(
-                        'failover.call_remote', 'cache.put', ['failover_encryption_keys', keys]
-                    )
-                except Exception as e:
-                    await self.middleware.call('alert.oneshot_create', 'FailoverKeysSyncFailed', None)
-                    self.middleware.logger.error('Failed to sync keys with remote node: %s', str(e), exc_info=True)
-                else:
-                    await self.middleware.call('alert.oneshot_delete', 'FailoverKeysSyncFailed', None)
+
+            # Only sync keys if we're the MASTER controller
+            if (await self.middleware.call('failover.status')) == 'MASTER':
+
+                # make sure we can contact the other controller
                 try:
-                    kmip_keys = await self.middleware.call('kmip.kmip_memory_keys')
-                    await self.middleware.call(
-                        'failover.call_remote', 'kmip.update_memory_keys', [kmip_keys]
-                    )
-                except Exception as e:
-                    await self.middleware.call(
-                        'alert.oneshot_create', 'FailoverKMIPKeysSyncFailed', {'error': str(e)}
-                    )
-                    self.middleware.logger.error(
-                        'Failed to sync KMIP keys with remote node: %s', str(e), exc_info=True
-                    )
-                else:
-                    await self.middleware.call('alert.oneshot_delete', 'FailoverKMIPKeysSyncFailed', None)
+                    assert (await self.middleware.call('failover.call_remote', 'core.ping')) == 'pong'
+                except Exception:
+                    self.middleware.logger.error('Failed to contact standby node.', exc_info=True)
```

Comment:
We should log that this happened during syncing keys just for reference if we want to know why keys did not sync